cmake/app/boilerplate.cmake

```diff
@@ -64,6 +64,9 @@ May include isr_tables.c etc."
   )
 set_property(GLOBAL PROPERTY GENERATED_KERNEL_SOURCE_FILES "")
 
+set(ZEPHYR_THIS_IS_A_BUILD_DIRECTORY TRUE CACHE INTERNAL
```

Comment:
If you really want to check, then check for `ZEPHYR_BASE`, that is being set here:
https://github.com/zephyrproject-rtos/zephyr/blob/master/share/zephyr-package/cmake/ZephyrConfig.cmake

even BEFORE boilerplate is included.